src/main/java/net/dv8tion/jda/internal/interactions/command/CommandInteractionPayloadImpl.java

```diff
@@ -91,31 +92,39 @@ private void parseResolved(JDAImpl jda, DataObject resolveJson)
         EntityBuilder entityBuilder = jda.getEntityBuilder();
 
         resolveJson.optObject("users").ifPresent(users ->
-                users.keys().forEach(userId -> {
-                    DataObject userJson = users.getObject(userId);
-                    UserImpl userArg = entityBuilder.createUser(userJson);
-                    resolved.put(userArg.getIdLong(), userArg);
-                })
+            users.keys().forEach(userId -> {
+                DataObject userJson = users.getObject(userId);
+                UserImpl userArg = entityBuilder.createUser(userJson);
+                resolved.put(userArg.getIdLong(), userArg);
+            })
+        );
+
+        resolveJson.optObject("attachments").ifPresent(attachments ->
+            attachments.keys().forEach(id -> {
+                DataObject json = attachments.getObject(id);
+                Message.Attachment file = entityBuilder.createMessageAttachment(json);
+                resolveJson.put(id, file);
```

Comment:
```suggestion
                resolved.put(Long.parseLong(id), file);
```
And `case ATTACHMENT:` needs to be added at https://github.com/DV8FromTheWorld/JDA/blob/6cbd75a1921ebae152a2f1fbf22c1c34f3d1c11c/src/main/java/net/dv8tion/jda/api/interactions/commands/OptionMapping.java#L292 to make this functional
(Sorry for messing this up the first time round)